main.go

```diff
@@ -48,5 +48,9 @@ func main() {
 
 	cmd.HeritageConfigFilePath = pwd + "/barcelona.yml"
 
-	app.Run(os.Args)
+	err = app.Run(os.Args)
+
+	if len(os.Args) != 1 && err == nil {
+		cmd.AutoRefreshVaultToken(app)
+	}
```

Comment:
You should not put this here because the vault token is only used in the client package, where it is used to communicate in barcelona. Please instead perform the relog in the client package to refresh the token.